src/include/distributed/metadata_sync.h

```diff
@@ -115,14 +135,42 @@ extern void SyncNewColocationGroupToNodes(uint32 colocationId, int shardCount,
 										  Oid distributionColumnCollation);
 extern void SyncDeleteColocationGroupToNodes(uint32 colocationId);
 
+extern MetadataSyncContext * CreateMetadataSyncContext(List *nodeList, bool testMode);
+extern void EstablishAndSetMetadataSyncBareConnections(MetadataSyncContext *context);
+extern void EstablishAndSetMetadataSyncCoordinatedConnections(
+	MetadataSyncContext *context);
+extern void SetMetadataSyncNodesFromNodeList(MetadataSyncContext *context,
+											 List *nodeList);
+extern void ResetMetadataSyncMemoryContext(MetadataSyncContext *context);
+extern bool MetadataSyncCollectsCommands(MetadataSyncContext *context);
+extern void SendCommandListToActivatedNodes(MetadataSyncContext *context, List *commands);
+extern void SendCommandListToMetadataNodes(MetadataSyncContext *context, List *commands);
+extern void SendCommandListToSingleWorker(MetadataSyncContext *context, List *commands,
+										  int nodeIdx);
+
+extern void ActivateNodeList(MetadataSyncContext *context);
+extern int ActivateNode(char *nodeName, int nodePort);
+
+extern char * WorkerDropAllShellTablesCommand(bool singleTransaction);
+
+extern void SendNodeWideObjectsSyncCommands(MetadataSyncContext *context);
+extern void SendShellTableDeletionCommands(MetadataSyncContext *context);
+extern void SendMetadataDeletionCommands(MetadataSyncContext *context);
+extern void SendColocationMetadataCommands(MetadataSyncContext *context);
+extern void SendDependencyCreationCommands(MetadataSyncContext *context);
+extern void SendDistTableMetadataCommands(MetadataSyncContext *context);
+extern void SendDistObjectCommands(MetadataSyncContext *context);
+extern void SendInterTableRelationshipCommands(MetadataSyncContext *context);
+
 #define DELETE_ALL_NODES "DELETE FROM pg_dist_node"
 #define DELETE_ALL_PLACEMENTS "DELETE FROM pg_dist_placement"
 #define DELETE_ALL_SHARDS "DELETE FROM pg_dist_shard"
 #define DELETE_ALL_DISTRIBUTED_OBJECTS "DELETE FROM pg_catalog.pg_dist_object"
 #define DELETE_ALL_PARTITIONS "DELETE FROM pg_dist_partition"
 #define DELETE_ALL_COLOCATION "DELETE FROM pg_catalog.pg_dist_colocation"
 #define REMOVE_ALL_SHELL_TABLES_COMMAND \
```

Comment:
I'd typiclly suggest keeping the marco name same as the UDF name for ease of search